ocm_python_wrapper/cluster.py

```diff
@@ -262,26 +263,91 @@ def addon_info(self):
             self.addon_name
         ).to_dict()
 
-    def validate_addon_parameters(self, parameters):
+    def validate_and_update_addon_parameters(
+        self, user_parameters, use_api_defaults=True
```

Comment:
why `user_parameters` is mandatory? 
If I have addon that have all parameters with defaults, then the user can call this without any parameters and get the defaults.

Reply:
Could be set with default such as user_parameters=[], but it requires to check outside if parameters are not empty, but the results are the same (pass empty list). 
If the case you suggested happens, then `user_parameters=[]` and the values will fill with default values if possible.

An indication of `user_parameters` (also an empty list) is mandatory for:
1. Check and raise if addon doesn't accept parameters but user gave parameters with values
2. The opposite case than you suggested - if all addon parameters have defaults, but the user specified a value for a parameter; then we need `user_parameters` for the check if parameter value not specified by user (and has default) - only then use default.

Reply:
if `user_parameters` can be empty list then it is not mandatory

Reply:
Done